firefox-ios/Client/Frontend/Browser/BrowserViewController/State/BrowserViewControllerState.swift

```diff
@@ -476,10 +468,36 @@ struct BrowserViewControllerState: ScreenState, Equatable {
                 microsurveyState: MicrosurveyPromptState.reducer(state.microsurveyState, action)
                 )
         default:
-            return state
+            return defaultActionState(from: state, action: action)
         }
     }
 
+    private static func defaultActionState(from state: BrowserViewControllerState,
+                                           action: Action) -> BrowserViewControllerState {
+        let defaultState = defaultActionState(from: state)
+        return BrowserViewControllerState(
+            searchScreenState: defaultState.searchScreenState,
+            showDataClearanceFlow: defaultState.showDataClearanceFlow,
+            fakespotState: FakespotState.reducer(defaultState.fakespotState, action),
+            windowUUID: defaultState.windowUUID,
+            browserViewType: defaultState.browserViewType,
+            microsurveyState: MicrosurveyPromptState.reducer(defaultState.microsurveyState, action)
+        )
+    }
+
+    static func defaultActionState(from state: BrowserViewControllerState) -> BrowserViewControllerState {
+        // This method should be used only in `defaultActionState(from: ,action: )` since the default state for this
+        // state depends on the action as well
+        return BrowserViewControllerState(
+            searchScreenState: state.searchScreenState,
+            showDataClearanceFlow: state.showDataClearanceFlow,
+            fakespotState: state.fakespotState,
```

Comment:
As mentioned in another comment: we should call the Fakespot and Microsurvey reducers here and remove the private `defaultActionState` - which probably means that we will need to pass in the action as well.

Reply:
Yes true, but i think `StateType` shouldn't know about `Action` in general. i'm looking into finding another solution.